executor/aggregate.go

```diff
@@ -51,15 +51,32 @@ type baseHashAggWorker struct {
 	aggFuncs     []aggfuncs.AggFunc
 	maxChunkSize int
 	stats        *AggWorkerStat
+
+	memTracker *memory.Tracker
+	BInMap     *int // indicate there are 2^BInMap buckets in Golang Map.
 }
 
-func newBaseHashAggWorker(ctx sessionctx.Context, finishCh <-chan struct{}, aggFuncs []aggfuncs.AggFunc, maxChunkSize int) baseHashAggWorker {
-	return baseHashAggWorker{
+const (
+	// ref https://github.com/golang/go/blob/go1.15.6/src/reflect/type.go#L2162.
+	// defBucketMemoryUsage = bucketSize*(1+unsafe.Sizeof(string) + unsafe.Sizeof(slice))+2*ptrSize
+	defBucketMemoryUsage = 8*(1+16+24) + 16
```

Comment:
Will the bucket size changed by golang in the future?

Reply:
Maybe.

Reply:
Just for curious, is there any way to set it dynamically according to different golang version?

Reply:
We can check runtime.Version() to distinguish different Golang version, and using different estimation way.  But we need to implement the estimation way for every different Golang version. (Of course, major version is enough). 
Now Golang 1.13,1.14.1.15,1.16 uses the same map implement, so I think now we don't need to distinguish them.